zingolib/src/lightclient.rs

```diff
@@ -1355,6 +1376,25 @@ impl LightClient {
         // The top of the wallet
         let last_synced_height = self.wallet.last_synced_height().await;
 
+        self.ensure_witness_tree_not_above_wallet_blocks().await;
```

Comment:
Let's a add comment explaining the scenario that could lead to the witness tree being ahead of the wallet blocks.

Reply:
Done